internal/app/pull/cli.go

```diff
@@ -151,7 +153,7 @@ func NewCommand(fullName string, err *os.File, out *os.File, in *os.File) *cobra
 				Distinct: distinct,
 			}
 
-			puller := pull.NewPuller(plan, datasource, pullExporterFactory(out), tracer)
+			puller := pull.NewPullerParallel(plan, datasource, pullExporterFactory(out), tracer, parallel)
```

Comment:
Create the PullerParallel with `pull.NewPullerParallel` only if parallel > 1.

If parallel <= 1, the use the `pull.NewPuller` method.

Reply:
My bad, it's already done in the pull.NewPullerParallel